src/serverrpc.cpp

```diff
@@ -0,0 +1,145 @@
+/******************************************************************************\
+ * Copyright (c) 2021
+ *
+ * Author(s):
+ *  dtinth
+ *
+ ******************************************************************************
+ *
+ * This program is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU General Public License as published by the Free Software
+ * Foundation; either version 2 of the License, or (at your option) any later
+ * version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU General Public License for more
+ * details.
+ *
+ * You should have received a copy of the GNU General Public License along with
+ * this program; if not, write to the Free Software Foundation, Inc.,
+ * 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA
+ *
+ \******************************************************************************/
+
+#include "serverrpc.h"
+
+CServerRpc::CServerRpc ( QObject* parent, CServer* pServer, CRpcServer* pRpcServer ) : QObject ( parent )
+{
+    pRpcServer->HandleMethod ( "jamulus/getMode", [=] ( const QJsonObject& params, QJsonObject& response ) {
+        QJsonObject result{ { "mode", "server" } };
+        response["result"] = result;
+        Q_UNUSED ( params );
+    } );
+
+    pRpcServer->HandleMethod ( "jamulusserver/getServerStatus", [=] ( const QJsonObject& params, QJsonObject& response ) {
```

Comment:
Any reason why this isn't split into recorder status and clients connected?
Wouldn't it be quicker -- especially for servers with many clients connected to have a 
jamulusserver/getAllClients
and
jamulusserver/getRecorderStatus

message instead of getServerStatus which has everything combined?

Reply:
Having done some first experiments with the API usage, I'm with @ann0see that we should split this. The client data is a lot more dynamic and the number of attributes might grow in the future. I think it'd be good to separate that.

What about simply `jamulusserver/getClients`?

Reply:
@hoffie No opinion; I put them all in a single method only because I was lazy :joy:. I have a cron job that calls `jamulusserver/getServerStatus` and sends everything to a PHP script that processes the JSON afterwards, this reduced my implementation effort on all sides.

Totally agree that splitting methods would provide a cleaner interface :+1:

Reply:
Thanks. Done in f6fa35179dc79d38c39986c011101b5bf4bddb76